pkg/controller/apimanager/apimanager_controller.go

```diff
@@ -0,0 +1,580 @@
+// a Pod as an example
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileAPIManager) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	r.reqLogger = log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	r.reqLogger.Info("Reconciling APIManager")
+
+	// Fetch the APIManager instance
+	instance := &appsv1alpha1.APIManager{}
+
+	r.reqLogger.Info("Trying to get APIManager resource")
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			r.reqLogger.Info("APIManager Resource not found. Ignoring since object must have been deleted")
+			return reconcile.Result{}, nil
+		}
+		r.reqLogger.Error(err, "APIManager Resource cannot be created. Requeuing request...")
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+	r.reqLogger.Info("Successfully retreived APIManager resource")
+
+	r.reqLogger.Info("Setting defaults for APIManager resource")
+	changed, err := instance.SetDefaults() // TODO check where to put this
+	if err != nil {
+		// Error setting defaults - Stop reconciliation
+		return reconcile.Result{}, nil
+	}
+	if changed {
+		r.reqLogger.Info("Updating defaults for APIManager resource")
+		err = r.client.Update(context.TODO(), instance)
+		if err != nil {
+			r.reqLogger.Error(err, "APIManager Resource cannot be updated. Requeuing request...")
+			return reconcile.Result{}, err
+		}
+		r.reqLogger.Info("Successfully updated defaults for APIManager resource")
+		return reconcile.Result{}, nil
+	}
+
+	objs, err := r.apiManagerObjects(instance)
+	if err != nil {
+		r.reqLogger.Error(err, "Error creating APIManager objects. Requeuing request...")
+		return reconcile.Result{}, err
+	}
+
+	// Set APIManager instance as the owner and controller
+	for idx := range objs {
+		obj := objs[idx]
+		obj.SetNamespace(instance.Namespace)
+		err = controllerutil.SetControllerReference(instance, obj, r.scheme)
+		if err != nil {
+			r.reqLogger.Error(err, "Error setting OwnerReference on object. Requeuing request...",
+				"Kind", obj.GetObjectKind(),
+				"Namespace", obj.GetNamespace(),
+				"Name", obj.GetName(),
+			)
+			return reconcile.Result{}, err
+		}
+	}
+
+	// Create APIManager Objects
+	for idx := range objs {
+		obj := objs[idx]
+		objectInfo := fmt.Sprintf("%s/%s", obj.GetObjectKind().GroupVersionKind().Kind, obj.GetName())
+
+		newobj := reflect.New(reflect.TypeOf(obj).Elem()).Interface()
+		found := newobj.(runtime.Object)
```

Comment:
Assuming all this is because of TypeMeta removal, it should no longer be needed with the use of KubeObject